src/engine/GroupByImpl.cpp

```diff
@@ -1458,7 +1471,7 @@ GroupByImpl::HashMapAggregationData<NUM_GROUP_COLUMNS>::getHashEntries(
     ++idx;
   }
 
-  return GroupLookupResult{std::move(matchedRowsToGroups),
+  return GroupLookupResult{std::move(matchedRows), std::move(groupIndexes),
```

Comment:
Any particular reason why this has changed (can't you still couple the matched and the groupIndexes?